Marlin/gcode.h

```diff
@@ -148,8 +147,7 @@ class GCodeParser {
 
     // Code is found in the string. If not found, value_ptr is unchanged.
     // This allows "if (seen('A')||seen('B'))" to use the last-found value.
-    // This is volatile because its side-effects are important
-    static volatile bool seen(const char c) {
+    static bool seen(const char& c) {
```

Comment:
> we're hinting the compiler that he can pass the memory address directly

A `char` is passed on the stack as 1 byte. A memory address will be 2 or 4 depending on architecture. Passing a char is more economical.

Hmm… I'm not sure why `volatile` is being discarded. Maybe it's redundant. Anyway, it is important that the function call always be a function call, and not get optimized away. I'll check the C++ docs to see whether a modifier is needed.